controllers/clustercache/cluster_cache.go

```diff
@@ -166,6 +166,12 @@ type ClusterCache interface {
 	GetClusterSource(controllerName string, mapFunc func(ctx context.Context, cluster client.Object) []ctrl.Request, opts ...GetClusterSourceOption) source.Source
 }
 
+// UncachedClientProvider provides live (uncached) access to the workload cluster API server.
+// This is an optional extension interface; callers can type assert a ClusterCache to this interface to retrieve an uncached client for a given workload cluster.
+type UncachedClientProvider interface {
+	GetUncachedClient(ctx context.Context, cluster client.ObjectKey) (client.Client, error)
```

Comment:
Let's add this to the existing `ClusterCache` interface instead. I don't want to introduce a new interface for every new method that we're adding

Reply:
I considered this. But if we do this, then won't it be a breaking Go API change? Whoever has implemented/used ClusterCache has to adjust no? (including tests, backports). Additionally, if most people don't need uncached access, then the main interface feels bloated.